### PR TITLE
chore(workflows): adding github workflows

### PR DESCRIPTION
**Pull request by schafle**
## PULL REQUEST TITLE
Adding multiple github workflows for gqltools

- [x] Enhancement
- [ ] Major Feature update
- [ ] Bug fixes 
- [ ] Breaking Change
- [ ] Documentation

## Added features and functionality
Workflow to lint, test, build and release gql tool CLI. Also added dependabot. 

## Issues resolved
NA

## Other
NA

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v3
+
+      - uses: actions/setup-go@v3
+        with:
+          go-version: 1.18
+
+      - name: Get Go dependencies
```
Comment (dylan-bourque) [inline .github/workflows/build.yml]: I don't think this step is necessary.  The `go test` command should download dependencies automatically and I don't have it in either `csproto` or `perseus`.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        run: go get -v -t -d ./...
+
+      - name: test
+        run: go test -v -race ./...
+
+      - name: Run golangci-lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          args: --timeout=3m12s
```
Comment (dylan-bourque) [inline .github/workflows/build.yml]: This seems oddly specific.  Why have a timeout of 3 minutes and 12 seconds? 😕

File: .github/workflows/gorelease-check.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+name: goreleaser-check
```
Comment (dylan-bourque) [inline .github/workflows/gorelease-check.yml:1]: 👍 for this.  Definitely going to add it to my 2 OSS repos.

**Review by dylan-bourque — APPROVED** [review]
A couple of comments/questions but LGTM

**Review by sonamp — APPROVED** [review]

**Pull request merged**